### PR TITLE
feat: add docker-buildx to Docker image

### PR DESCRIPTION
**Pull request by msvechla**
This commit will add the `buildx` docker plugin, to enable multi-arch Docker builds.
For more information see #2332.

Let me know what you think, whether this makes sense, or if some modifications are required.
Thanks a lot!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,3 +16,7 @@ RUN chmod +x /entrypoint.sh
 
 COPY goreleaser_*.apk /tmp/
 RUN apk add --allow-untrusted /tmp/goreleaser_*.apk
+
+RUN mkdir -p $HOME/.docker/cli-plugins/ && \
```
Comment (caarlos0) [inline Dockerfile:20]: I think we can just add `docker-cli-buildx` to the list of packages to install in line 3...

Comment (msvechla) [reply Dockerfile:20]: even better, I will give it a try and update the PR if it works. Thanks for the hint!

Comment (msvechla) [reply Dockerfile]: works great, I updated the PR

**Comment by crazy-max** [comment]
I don't think it will work as intended because you need a buildkit container to allow building against other platforms (through [`docker-container`](https://github.com/docker/buildx/blob/master/docs/reference/buildx_create.md#driver) driver).

File: Dockerfile (hunk 1 above)
Comment (crazy-max) [reply Dockerfile]: You can also use our [buildx-bin](https://github.com/docker/buildx#from-dockerfile) image to add buildx to your image: https://github.com/docker/buildx#from-dockerfile

**Comment by msvechla** [comment]
Yes, this is not the only configuration required, but this is the only configuration required from the `goreleaser` docker image. You can see my successful pipeline run here: https://gitlab.com/msvechla/vaultbot/-/jobs/1399876389 

There I use my custom image of `goreleaser` with these changes applied.
I also had to do some other things to prepare the linked service DinD environment, but it is working.

This is the `.gitlab-ci.yml` file I am using:

```release:
  stage: release
  image: docker:stable
  before_script:
    - mkdir -p $HOME/.docker/cli-plugins/
    - wget -O $HOME/.docker/cli-plugins/docker-buildx $BUILDX_URL
    - chmod a+x $HOME/.docker/cli-plugins/docker-buildx
    - "echo -e '{\n  \"experimental\": \"enabled\"\n}' | tee $HOME/.docker/config.json"
    - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
    - docker buildx create --use --driver docker-container --name multiarch --platform=${BUILDX_PLATFORM}
  services:
    - docker:dind

  variables:
    # Disable shallow cloning so that goreleaser can diff between tags to
    # generate a changelog.
    GIT_DEPTH: 0

  script: |
    rm -f gl-code-quality-report.json
    docker run --rm --privileged \
      -v $PWD:/go/src/gitlab.com/msvechla/vaultbot \
      -w /go/src/gitlab.com/msvechla/vaultbot \
      -v /var/run/docker.sock:/var/run/docker.sock \
      -e DOCKER_USERNAME -e DOCKER_PASSWORD -e DOCKER_REGISTRY  \
      -e GITLAB_TOKEN \
      msvechla/goreleaser:latest release --rm-dist
  only:
    - tags
```

**Comment by caarlos0** [comment]
I think you are probably using the host as the DOCKER_HOST, so probably it has the builder stuff ready.

I believe this should be fine.

Thanks for the PR! 🙏

**Review by caarlos0 — APPROVED** [review]

**Pull request merged**